src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -5902,6 +5902,16 @@ and TcExprUndelayed (cenv: cenv) (overallTy: OverallTy) env tpenv (synExpr: SynE
             TcPossiblyPropagatingExprLeafThenConvert (fun ty -> isAnonRecdTy g ty || isTyparTy g ty) cenv overallTy env mWholeExpr (fun overallTy ->
                 TcAnonRecdExpr cenv overallTy env tpenv (isStruct, withExprOpt, unsortedFieldExprs, mWholeExpr)
             )
+        | Some(SynExpr.LongIdent (_, lId, _, _), _) as Some withExpr ->
```

Comment:
This looks somewhat ad-hoc. I think it would be better to find a better implementation for the code above.